src/utils/slot.py

```diff
@@ -0,0 +1,138 @@
+import logging
+from http import HTTPStatus
+from typing import Optional
+
+from src.providers.consensus.client import ConsensusClient
+from src.providers.consensus.typings import BlockHeaderFullResponse
+from src.providers.http_provider import NotOkResponse
+from src.typings import BlockStamp, SlotNumber, EpochNumber, BlockNumber
+
+
+logger = logging.getLogger(__name__)
+
+
+class NoSlotsAvailable(Exception):
+    pass
+
+
+class InconsistentData(Exception):
+    pass
+
+
+class SlotNotFinalized(Exception):
+    pass
+
+
+def get_first_non_missed_slot(
+    cc: ConsensusClient,
+    ref_slot: SlotNumber,
+    last_finalized_slot_number: SlotNumber,
+    ref_epoch: Optional[EpochNumber] = None,
+) -> BlockStamp:
```

Comment:
We use this not only with `ref_slot`, in bunker for example. And since this method returns blockstamp with ref_slot and ref_epoch this can cause issues.

I would suggest splitting it into two methods. One finds the closest block to the slot, the other expands blockstamp with additional fields: ref_slot and ref_epoch